src/ngraph/runtime/gpu/cudnn_emitter.cpp

```diff
@@ -1403,6 +1403,69 @@ size_t runtime::gpu::CUDNNEmitter::build_batchnorm(const cudnnBatchNormMode_t& b
     return this->m_primitive_emitter->register_primitive(batchnorm, hash);
 }
 
+size_t runtime::gpu::CUDNNEmitter::build_lrn(const cudnnLRNMode_t& lrn_op,
+                                             const std::string& dtype,
+                                             const Prop& direction,
+                                             const Shape& input_shape,
+                                             const Shape& output_shape,
+                                             const double lrn_alpha,
+                                             const double lrn_beta,
+                                             const double lrn_bias,
+                                             const size_t lrn_size)
+{
+    // construct hash to determine if kernel needs to be emitted
+    // or if it already exists in the primitive list
+    std::stringstream ss;
+    ss << "lrn_op_" << lrn_op << "_dtype_" << dtype << "_dir" << static_cast<int>(direction) << "_i"
```

Comment:
```
                                              const double lrn_alpha,
                                              const double lrn_beta,
                                              const double lrn_bias,
                                              const size_t lrn_size
```
Those should also in hash. Since here there is not runtime input for those numbers. So different number can not share the same primitive. 

Another way is put those number through parameter like `void** inputs, void** outputs`, so you don't need those number in hash. This require some framework change.

Reply:
I'll add these to the hash (at least for now). That will be consistent with doing cudnnSetLRNDescriptor once ahead of time (previous comment above). (If passed as param, I would not do cudnnSetLRNDescriptor ahead of time). But agreed that it would likely be a good future direction to pass additional info as parameter (generating less unique ops) to favor faster compilation (because runtime would likely not be adversely impacted).